image/config.go

```diff
@@ -73,37 +73,41 @@ func (c *config) runtimeSpec(rootfs string) (*specs.Spec, error) {
 	var s specs.Spec
 	s.Version = specs.Version
 	// we should at least apply the default spec, otherwise this is totally useless
-	s.Process.Terminal = true
-	s.Root.Path = rootfs
-	s.Process.Cwd = "/"
-	if c.Config.WorkingDir != "" {
-		s.Process.Cwd = c.Config.WorkingDir
+	if s.Root != nil {
+		s.Root.Path = rootfs
 	}
-	s.Process.Env = append(s.Process.Env, c.Config.Env...)
-	s.Process.Args = append(s.Process.Args, c.Config.Entrypoint...)
-	s.Process.Args = append(s.Process.Args, c.Config.Cmd...)
-
-	if len(s.Process.Args) == 0 {
-		s.Process.Args = append(s.Process.Args, "sh")
-	}
-
-	if uid, err := strconv.Atoi(c.Config.User); err == nil {
-		s.Process.User.UID = uint32(uid)
-	} else if ug := strings.Split(c.Config.User, ":"); len(ug) == 2 {
-		uid, err := strconv.Atoi(ug[0])
-		if err != nil {
-			return nil, errors.New("config.User: unsupported uid format")
+	if s.Process != nil {
+		s.Process.Terminal = true
+		s.Process.Cwd = "/"
+		if c.Config.WorkingDir != "" {
+			s.Process.Cwd = c.Config.WorkingDir
 		}
+		s.Process.Env = append(s.Process.Env, c.Config.Env...)
+		s.Process.Args = append(s.Process.Args, c.Config.Entrypoint...)
+		s.Process.Args = append(s.Process.Args, c.Config.Cmd...)
 
-		gid, err := strconv.Atoi(ug[1])
-		if err != nil {
-			return nil, errors.New("config.User: unsupported gid format")
+		if len(s.Process.Args) == 0 {
+			s.Process.Args = append(s.Process.Args, "sh")
```

Comment:
@stevvooe If there are no other questions, can we merge this patch first and then discuss the `sh` problem? Because if you don't merge this, the `create` function won't work.